vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/HttpRequestImpl.java

```diff
@@ -131,6 +133,17 @@ public MultiMap headers() {
     return headers;
   }
 
+  @Override
+  public HttpRequest<T> setHost(String host) {
```

Comment:
we should use the same setter convention than the class and have `host(String)` instead

Reply:
Well, there is already host(String) method in the interface used to configure the host the request need to connect to: https://github.com/vert-x3/vertx-web/blob/master/vertx-web-client/src/main/java/io/vertx/ext/web/client/HttpRequest.java#L96
So, maybe instead `setHost(String)` it should be `hostHeader(String)`, what do you think ?

Reply:
so in this case, one can set the host header and it will behave the same way

Reply:
In fact, "host" *header* does not have any effect, check this: https://github.com/eclipse/vert.x/blob/master/src/main/java/io/vertx/core/http/impl/HttpClientRequestImpl.java#L644 it respects only `hostHeader` field being set by setHost(). You can't do the same by doing e.g. `request.headers().put("host","other-host");`

Maybe this is the real problem, that underlying httpclientrequest implemention does not respect `headers().get('host')` but instead new field.